### PR TITLE
update profile doc API; add instrucitons for GCP access fine grain control

### PR DESCRIPTION
**Pull request by kunmingg**
fix https://github.com/kubeflow/website/issues/1313

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1318)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
Also addresses issue https://github.com/kubeflow/website/issues/1283

File: content/docs/other-guides/multi-user-overview.md [hunk 1]
```diff
@@ -151,6 +151,32 @@ The following resources are created as part of the profile creation:
 Namespaces, Profiles and Namespaces are sometimes used interchangably in the
 documentation.
 
+### Cloud API Authenticate with Profile
+
+**GCP**: User profile on GCP will be integrated with [GKE Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+User pods will be authenticated automatically when talk to GCP APIs.
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: "User pods will be authenticated automatically when talk to GCP APIs." -> 
"With workload identity, GCP credentials can be assigned to a pod by binding a Google Service Account to the Kubernetes Service Account used by the pod."

File: content/docs/other-guides/multi-user-overview.md [hunk 2]
```diff
@@ -151,6 +151,32 @@ The following resources are created as part of the profile creation:
...
+
+By default all profiles in a cluster will share a public GCP service account identity.
```
Comment (jlewi) [inline content/docs/other-guides/multi-user-overview.md]: Could we clarify what is meant by default here?
I assume this is only true if user deployed with kfctl_gcp_iap.yaml or if their KFDef contains the GCPPluginSpec.

So we could say something like that e..g

"If you deployed Kubeflow following the GCP instructions then the profiler controller will automatically bind the "default-editor" service account to the GCP service account".

**Comment by jlewi** [comment]
@kunmingg think we can another revision done today?

**Comment by sarahmaddox** [comment]
Preview: https://deploy-preview-1318--competent-brattain-de2d6d.netlify.com/docs/gke/authentication/#starting-in-0-7-gke-workload-identity

/assign @jlewi 

@kunmingg thanks for these updates! @jlewi I'll leave the review to you, as you're doing such a great job of it already. :)

File: content/docs/gke/authentication.md [hunk 3]
```diff
@@ -52,6 +53,52 @@ gcloud --project=${PROJECT} iam service-accounts get-iam-policy ${KFNAME}-user@$
...
+    name: user1@email.com   # replace with the email of the user
+  plugins:
+  - kind: WorkloadIdentity
+    spec:
+      gcpServiceAccount: user1-gcp@project-id.iam.gserviceaccount.com
+```
+
+**Note:**
+Profile controller currently doesn't perform any access control check to see whether the user creating the profile should be able to use the GCP service account. 
+So any user who can create a profile can get access to any service account for which the admin controller has owner permissions on. We will improve this in subsequent releases and link to an issue.
```
Comment (jlewi) [inline content/docs/gke/authentication.md]: Either delete "link to an issue" or replace it with an actual link to an issue

Comment (kunmingg) [reply content/docs/gke/authentication.md]: sorry, updated.

**Pull request merged**